File: 0938-range-sum-of-bst/0938-range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
           int ans= inorder(root, 0, low, high);
        return ans;
    }
    int inorder(TreeNode root, int sum, int l, int h){
        if(root== null){
            return 0;
        }
        int s1= inorder(root.left, sum, l, h);
        int s2= inorder(root.right, sum, l, h);
        int x= root.val<=h && root.val>=l ? root.val: 0;
        return s1+s2+ x;
    }
}

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def rangeSumBST(root: TreeNode, low: Int, high: Int): Int = {
        def getSumInRange(rt: TreeNode, l: Int, h: Int): Int={
            if(rt== null){
                0
            }
            else if(rt.value>= l && rt.value<= h){
                val sl= getSumInRange(rt.left, l, h) 
                val sr= getSumInRange(rt.right, l, h)
                rt.value+ sl+ sr
            }
            else{
                val sl= getSumInRange(rt.left, l, h) 
                val sr= getSumInRange(rt.right, l, h)
                sl+ sr
            }
            
        }  
        getSumInRange(root, low, high)
    }
}